lib/discourse_api/client.rb

```diff
@@ -100,6 +100,8 @@ def connection
       @connection ||= Faraday.new connection_options do |conn|
         # Follow redirects
         conn.use FaradayMiddleware::FollowRedirects, limit: 5
+        # Allow uploading of files
+        conn.request :multipart
```

Comment:
Okay sounds good. Looks like the docs say "Request::Multipart checks for files in the payload, otherwise leaves everything untouched;" so I guess it is okay to leave in for all requests.